src/main/java/com/exactpro/th2/conn/MicroserviceMain.java

```diff
@@ -284,7 +309,24 @@ private static void createPackAndPublishPipeline(Direction direction, Flowable<C
                 .subscribe(batch -> {
                     try {
                         RawMessageBatch rawBatch = batch.convertToProtoRawBatch();
-                        rawMessageRouter.sendAll(rawBatch, (direction == Direction.FIRST ? QueueAttribute.FIRST : QueueAttribute.SECOND).toString());
+                        RawMessage firstMessage = rawBatch.getMessages(0);
+                        String firstMessageBookName = firstMessage.getMetadata().getId().getBookName();
+                        String parentEventIdBookName = firstMessage.getParentEventId().getBookName();
+                        if (!Objects.equals(firstMessageBookName, parentEventIdBookName)) {
+                            eventDispatcher.store(EventHolder.createError(Event
+                                    .start()
+                                    .endTimestamp()
+                                    .name(String.format(
+                                            "Message in raw batch has bookName `%s but parentEventId book name is `%s`",
+                                            firstMessageBookName,
+                                            parentEventIdBookName
+                                    ))
+                                    .status(Event.Status.FAILED)
+                                    .type("Error")
+                            ));
```

Comment:
This check is redundant because the MessageSender must reject incoming messages not matched by book name and the ServiceListener must use book name from the common factory.

Reply:
done